include/deal.II/base/mpi.h

```diff
@@ -1539,11 +1539,195 @@ namespace Utilities
+    {
+      return _builtin_broadcast(comm, object_to_send, root_process);
+    }
+
+
+
+    inline long int
+    broadcast(const MPI_Comm    &comm,
+              const long int    &object_to_send,
+              const unsigned int root_process)
+    {
+      return _builtin_broadcast(comm, object_to_send, root_process);
+    }
+
+
+
+    inline unsigned char
+    broadcast(const MPI_Comm      &comm,
+              const unsigned char &object_to_send,
+              const unsigned int   root_process)
+    {
+      return _builtin_broadcast(comm, object_to_send, root_process);
+    }
+
+
+
+    inline unsigned short
+    broadcast(const MPI_Comm       &comm,
+              const unsigned short &object_to_send,
+              const unsigned int    root_process)
+    {
+      return _builtin_broadcast(comm, object_to_send, root_process);
+    }
+
+
+
+    inline unsigned int
+    broadcast(const MPI_Comm     &comm,
+              const unsigned int &object_to_send,
+              const unsigned int  root_process)
+    {
+      return _builtin_broadcast(comm, object_to_send, root_process);
+    }
+
+
+
+    inline unsigned long int
+    broadcast(const MPI_Comm          &comm,
+              const unsigned long int &object_to_send,
+              const unsigned int       root_process)
+    {
+      return _builtin_broadcast(comm, object_to_send, root_process);
+    }
+
+
+
+    inline unsigned long long int
+    broadcast(const MPI_Comm               &comm,
+              const unsigned long long int &object_to_send,
+              const unsigned int            root_process)
+    {
+      return _builtin_broadcast(comm, object_to_send, root_process);
+    }
+
+
+
+    inline float
+    broadcast(const MPI_Comm    &comm,
+              const float       &object_to_send,
+              const unsigned int root_process)
+    {
+      return _builtin_broadcast(comm, object_to_send, root_process);
+    }
+
+
+
+    inline double
+    broadcast(const MPI_Comm    &comm,
+              const double      &object_to_send,
+              const unsigned int root_process)
+    {
+      return _builtin_broadcast(comm, object_to_send, root_process);
+    }
+
+
+
+    inline long double
+    broadcast(const MPI_Comm    &comm,
+              const long double &object_to_send,
+              const unsigned int root_process)
+    {
+      return _builtin_broadcast(comm, object_to_send, root_process);
+    }
+
+
+
+    inline std::complex<float>
+    broadcast(const MPI_Comm            &comm,
+              const std::complex<float> &object_to_send,
+              const unsigned int         root_process)
+    {
+      return _builtin_broadcast(comm, object_to_send, root_process);
+    }
+
+    inline std::complex<double>
+    broadcast(const MPI_Comm             &comm,
+              const std::complex<double> &object_to_send,
+              const unsigned int          root_process)
+    {
+      return _builtin_broadcast(comm, object_to_send, root_process);
+    }
+
```

Comment:
Just replace all overloads with
```suggestion
    template <typename T>
    std::enable_if_t<is_mpi_type<T>::value,T> 
    broadcast(const MPI_Comm    &comm,
              const T           &object_to_send,
              const unsigned int root_process)
    {
      return _builtin_broadcast(comm, ArrayView<T>(&object_to_send, 1), root_process);
    }
```
and possibly another overload taking an `ArrayView`.

Reply:
Fixed in 5a80fb6cc8996315c5328863c6f4df602a8efbe6